### PR TITLE
Cleanup README for clippy-preview on stable

### LINKED ISSUE #3177: Clippy now works in Rust stable (1.29): update docs accordingly?
According to the [Rust 1.29 announcement](https://blog.rust-lang.org/2018/09/13/Rust-1.29.html), Clippy now works with stable Rust.

### PR DESCRIPTION
**Pull request by phansch**
With the 1.29 release, the `clippy-preview` component will be available
on stable which means we don't need nightly/beta anymore.

This also removes the `rust-update` script which was still using `cargo install` to install clippy.

Closes #3177

### CODE REVIEW

**Comment by phansch** [comment]
We can merge this in 2 days with the next stable release :rocket:

File: README.md [hunk 1]
```diff
@@ -54,26 +54,15 @@ rustup and compiler:
 rustup update
 ```
 
-#### Step 2: Install nightly toolchain
+#### Step 2: Install clippy
```
Comment (mati865) [inline README.md]: Pedantic: It's now called `Clippy`.

**Comment by phansch** [comment]
Going to give this a try and then merge :balloon:

**Comment by phansch** [comment]
:fireworks: :fireworks: :fireworks:

**Pull request merged**